### PR TITLE
[BUGFIX release] unregister views from viewRegistry (release)

### LINKED ISSUE #14526: ember 2.9 2.8 not unregistering views correctly
2.9.0
* https://github.com/emberjs/ember.js/blob/v2.9.0/packages/ember-htmlbars/lib/renderer.js#L263
* https://github.com/emberjs/ember.js/blob/v2.9.0/packages/ember-glimmer/lib/renderer.js#L226

2.8.1
* https://github.com/emberjs/ember.js/blob/v2.8.1/packages/ember-htmlbars/lib/renderer.js#L263
* https://github.com/emberjs/ember.js/blob/v2.8.1/packages/ember-glimmer/lib/renderer.js#L226

2.8.0
* https://github.com/emberjs/ember.js/blob/v2.8.0/packages/ember-htmlbars/lib/renderer.js#L263
* https://github.com/emberjs/ember.js/blob/v2.8.1/packages/ember-glimmer/lib/renderer.js#L226

introduced `v2.8.0-beta.1` https://github.com/emberjs/ember.js/commit/2a233df8c6aa387898299d79dd4b2965b4266015

--

Root cause really that the original code appeared to not be covered by tests, so reasonable refactoring broke it, but without test failure.

--

issue to be sure we don't forget about a test on master: https://github.com/emberjs/ember.js/issues/14530

### PR DESCRIPTION
**Pull request by stefanpenner**
[FIXES #14526]
- [x] ~~needs test~~ (tests added upstream in https://github.com/emberjs/ember.js/pull/14100)

### CODE REVIEW

File: packages/ember-glimmer/lib/renderer.js [hunk 1]
```diff
@@ -218,12 +218,12 @@ class Renderer {
   }
 
   _register(view) {
-    assert('Attempted to register a view with an id already in use: ' + view.elementId, !this._viewRegistry[this.elementId]);
+    assert('Attempted to register a view with an id already in use: ' + view.elementId, !view._viewRegistry[this.elementId]);
```
Comment (sly7-7) [inline packages/ember-glimmer/lib/renderer.js]: `this._viewRegistry` and `view.elementId` right ?

**Comment by rwjblue** [comment]
This should target `release` branch.

**Comment by rwjblue** [comment]
(updated via GH UI)

**Comment by rwjblue** [comment]
Rebased on top of https://github.com/emberjs/ember.js/pull/14531 (to fix the JSHint errors this was having).

**Comment by stefanpenner** [comment]
thanks for driving this to completion @rwjblue i totally just fell asleep.

**Pull request merged**